### PR TITLE
Feature/condo/doma 4545/ticket search

### PR DESCRIPTION
**Pull request by Alllex202**

### CODE REVIEW

File: apps/condo/domains/ticket/schema/Ticket.js [hunk 1]
```diff
@@ -541,6 +561,15 @@ const Ticket = new GQLListSchema('Ticket', {
         delete: false,
         auth: true,
     },
+    kmigratorOptions: {
+        indexes: [
+            {
+                type: 'Index', // or BTreeIndex
```
Comment (pahaz) [inline apps/condo/domains/ticket/schema/Ticket.js:567]: it looks lika part of another PR

File: apps/condo/domains/user/schema/User.js [hunk 2]
```diff
@@ -214,6 +215,7 @@ const User = new GQLListSchema('User', {
         dvAndSender(),
         historical(),
         webHooked(),
+        searchUpdateByTrigger([{ targetSchema: 'Ticket', triggerPathsToFields: ['assignee.name', 'executor.name'], schemaGql: Ticket }]),
```
Comment (pahaz) [inline apps/condo/domains/user/schema/User.js:218]: it looks like a bad code design. Probably this should work automatically inside the `searchBy` plugin

File: packages/keystone/plugins/searchBy.js [hunk 3]
```diff
@@ -0,0 +1,189 @@
...
+    kmigratorOptions: { null: false },
+    defaultValue: SEARCH_FIELD_DEFAULT_VALUE,
+    access: {
+        create: false,
+        read: true,
+        update: true,
+    },
+    knexOptions: { defaultTo: () => SEARCH_FIELD_DEFAULT_VALUE },
+}
+const SEARCH_VERSION_OPTIONS = {
```
Comment (pahaz) [inline packages/keystone/plugins/searchBy.js:23]: how is it supposed to be used ?

File: packages/keystone/plugins/searchBy.js [hunk 4]
```diff
@@ -0,0 +1,189 @@
...
+    schemaDoc: 'Service field for versioning the search field by different fields',
+    type: 'Integer',
+    isRequired: true,
+    defaultValue: 1,
+    access: {
+        create: false,
+        read: true,
+        update: true,
```
Comment (pahaz) [inline packages/keystone/plugins/searchBy.js:31]: anyone can update this field?

File: packages/keystone/plugins/searchBy.js [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+
+const SEARCH_OPTIONS = {
+    schemaDoc: 'Service field for searching by different fields',
+    type: 'Text',
+    kmigratorOptions: { null: false },
+    defaultValue: SEARCH_FIELD_DEFAULT_VALUE,
+    access: {
+        create: false,
+        read: true,
+        update: true,
```
Comment (pahaz) [inline packages/keystone/plugins/searchBy.js:19]: anyone can update this field

Comment (Alllex202) [reply packages/keystone/plugins/searchBy.js:19]: no one

File: packages/keystone/plugins/searchBy.js (hunk 4 above)
Comment (Alllex202) [reply packages/keystone/plugins/searchBy.js:31]: only server utils

File: apps/condo/domains/user/schema/User.js (hunk 2 above)
Comment (Alllex202) [reply apps/condo/domains/user/schema/User.js:218]: moved to global preprocessors

Comment (Alllex202) [reply apps/condo/domains/user/schema/User.js]: Moved to global plugins

File: packages/keystone/plugins/searchBy.js (hunk 3 above)
Comment (Alllex202) [reply packages/keystone/plugins/searchBy.js]: The "searchVersion" field is used as a trigger.
This allows you to control the values of the "search" field from the main model. Linked models can only report that they have updated without interacting with the "search" field directly.

Tracked field updated --> update "searchVersion" field --> need to update "search" field.

Comment (Alllex202) [reply packages/keystone/plugins/searchBy.js]: But it is better to use not a number, but some random value. For example, uuid

**Comment by Alllex202** [comment]
Issue: updating related models can trigger a bulk ticket update

**Pull request closed without merging**